From adcaafcc6c86c079c682d0c9a79ddfa9996ff989 Mon Sep 17 00:00:00 2001
From: Juergen Beisert <jbe@pengutronix.de>
Date: Thu, 11 Mar 2010 17:26:01 +0100
Subject: [PATCH 54/59] imx: keep the lcd clock running

To avoid any display picture fading at system startup, keep the LCD
clock running. This works due to the fact that the booloader initializes
the LCD already. So the display shows a blank screen until the in-kernel
LCD driver takes over.

Signed-off-by: Juergen Beisert <jbe@pengutronix.de>
---
 arch/arm/mach-mx2/clock_imx27.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: linux-2.6.33/arch/arm/mach-mx2/clock_imx27.c
===================================================================
--- linux-2.6.33.orig/arch/arm/mach-mx2/clock_imx27.c
+++ linux-2.6.33/arch/arm/mach-mx2/clock_imx27.c
@@ -742,7 +742,7 @@ int __init mx27_clocks_init(unsigned lon
 		clkdev_add(&lookups[i]);
 
 	/* Turn off all clocks we do not need */
-	__raw_writel(0, CCM_PCCR0);
+	__raw_writel((1 << 14), CCM_PCCR0);	/* keep the LCD clock running */
 	__raw_writel((1 << 10) | (1 << 19), CCM_PCCR1);
 
 	spll_clk.disable(&spll_clk);
Index: linux-2.6.33/arch/arm/mach-mx25/clock.c
===================================================================
--- linux-2.6.33.orig/arch/arm/mach-mx25/clock.c
+++ linux-2.6.33/arch/arm/mach-mx25/clock.c
@@ -225,10 +225,10 @@ int __init mx25_clocks_init(void)
 
 	/* Turn off all clocks except the ones we need to survive, namely:
 	 * EMI, GPIO1-3 (CCM_CGCR1[18:16]), GPT1, IOMUXC (CCM_CGCR1[27]), IIM,
-	 * SCC
+	 * SCC, LCD, PWM (backlight)
 	 */
-	__raw_writel((1 << 19), CRM_BASE + CCM_CGCR0);
-	__raw_writel((0xf << 16) | (3 << 26), CRM_BASE + CCM_CGCR1);
+	__raw_writel((1 << 19) | (1 << 7) | (1 << 24) | (1 << 10), CRM_BASE + CCM_CGCR0);
+	__raw_writel((0xf << 16) | (3 << 26) | (1 << 29), CRM_BASE + CCM_CGCR1);
 	__raw_writel((1 << 5), CRM_BASE + CCM_CGCR2);
 
 	mxc_timer_init(&gpt_clk, MX25_IO_ADDRESS(MX25_GPT1_BASE_ADDR), 54);
